From 1814a8de09980d7ae7f0d74a16bb3c10c3194165 Mon Sep 17 00:00:00 2001
From: Amelie Delaunay <amelie.delaunay@foss.st.com>
Date: Thu, 15 Dec 2022 15:52:14 +0100
Subject: [PATCH 0099/1141] dt-bindings: dma: stm32-dma3: prevent
 packing/unpacking mode

When source data width/burst and destination data width/burst are
different, data are packed or unpacked in DMA3 channel FIFO, using
CxTR1.PAM.
Data are pushed out from DMA3 channel FIFO when the destination burst
length (= data width * burst) is reached.
If the transfer is stopped before CxBR1.BNDT = 0, and if some bytes are
packed/unpacked in the DMA3 channel FIFO, these bytes are lost.
DMA3 channel FIFO has no flush capability, only reset.
To avoid potential bytes lost, pack/unpack must be prevented by setting
memory data width/burst equal to peripheral data width/burst.
Memory accesses will be penalized. But it is the only way to avoid bytes
lost.

Prevent pack/unpack feature can be activated by setting bit 16 of DMA3
Transfer requirements bitfield (tr_conf).

Signed-off-by: Amelie Delaunay <amelie.delaunay@foss.st.com>
Change-Id: I4a5c0080e9875c5306574c30414f076e0de70d74
---
 Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml | 3 +++
 1 file changed, 3 insertions(+)

diff --git a/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml b/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
index 5f5917bfc28b..e7c756f16143 100644
--- a/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
+++ b/Documentation/devicetree/bindings/dma/st,stm32-dma3.yaml
@@ -57,6 +57,9 @@ description: |
         0x2: at LLI level, the transfer complete event is generated at the end of the LLI transfer,
              including the update of the LLI if any
         0x3: at channel level, the transfer complete event is generated at the end of the last LLI
+      -bit 16: Prevent packing/unpacking mode
+        0x0: pack/unpack enabled when source data width/burst != destination data width/burst
+        0x1: memory data width/burst forced to peripheral data width/burst to prevent pack/unpack
     4. A 32-bit mask specifying the DMA transfer extended requirements
       -bit 0-12: The source address offset increment
       -bit 13: The source address increment type
-- 
2.39.2

